### DISCUSSION TITLE
settings namings should follow django style

### DISCUSSION THREAD

**Discussion by newearthmartin** [celery/celery#7038] [Ideas]
Celery 5 breaks Django's convention of having CAPITAL_LETTER_NAMES in settings. Currently, Celery settings are the only lowercase settings in my whole settings.py file. Also by removing the CELERY_ prefix, it's not easy to know that these settings belong to celery.

For example, there is the Django `TIME_ZONE` setting and then celery adds a `timezone` setting. It would be  respectful if Celery could mark these names as CELERY_. **Imagine if all apps started doing what Celery did, how would the settings file look like?**

[a comment by an automated account is not shown]

**Comment by auvipy** [comment]
the settings were changed in version 4, v5 just removed the deprecated old settings

**Comment by auvipy** [comment]
anyway I am closing this for now. if any future consensus is reached to rechange it, will reopen.

**Comment by newearthmartin** [comment]
I guess it's not going to look good to go back to the pre v4 settings.
I am concerned that this breaks the convention for Django settings. And as I said it does not play well with other apps. Imagine if all apps that use timezones added a plain `timezone` setting.

**Comment by newearthmartin** [comment]
All apps prefix their settings with the app name AFAIK. And all apps use uppercase. If there was consensus about that it would be great. It used to be the case with Celery and I'm not sure why it was changed in the first place.

**Comment by LincolnPuzey** [comment]
@newearthmartin you can still use upper case name with a `CELERY_` prefix.

See https://docs.celeryproject.org/en/stable/history/whatsnew-4.0.html#latentcall-django-admonition
and https://docs.celeryproject.org/en/stable/django/first-steps-with-django.html#using-celery-with-django

**Comment by odigity** [comment]
I'm still confused as to what I'm supposed to be doing.

The [release for 4.0](https://docs.celeryproject.org/en/stable/history/whatsnew-4.0.html#latentcall-django-admonition
) said:

    "If you’re loading Celery configuration from the Django settings module then you’ll want to keep using the uppercase names."

But the [current config docs](https://docs.celeryproject.org/en/stable/userguide/configuration.html#new-lowercase-settings) say:

    "Celery will still be able to read old configuration files until Celery 6.0. Afterwards, support for the old configuration files will be removed."

    "Please migrate to the new configuration scheme as soon as possible."

So, which is it?

**Reply by odigity** [reply]
After much code-reading and experimentation, I crafted this to aid me in the future:

[Celery Settings Primer](https://gist.github.com/odigity/35e36d350884bdaa57998a563b3a735a)

**Comment by newearthmartin** [comment]
These name changes for settings only make sense when you add the `namespace` setting. Wonder why documentation is not clear about this.

**Reply by odigity** [reply]
The only part of the docs I can find that even comes close to explaining this is on the [Django page](https://docs.celeryproject.org/en/stable/django/first-steps-with-django.html):

   > `app.config_from_object('django.conf:settings', namespace='CELERY')`
   >
   > The uppercase name-space means that all [Celery configuration options](https://docs.celeryproject.org/en/stable/userguide/configuration.html#configuration) must be specified in uppercase instead of lowercase, and start with CELERY_, so for example the [task_always_eager](https://docs.celeryproject.org/en/stable/userguide/configuration.html#std-setting-task_always_eager) setting becomes CELERY_TASK_ALWAYS_EAGER, and the [broker_url](https://docs.celeryproject.org/en/stable/userguide/configuration.html#std-setting-broker_url) setting becomes CELERY_BROKER_URL. This also applies to the workers settings, for instance, the [worker_concurrency](https://docs.celeryproject.org/en/stable/userguide/configuration.html#std-setting-worker_concurrency) setting becomes CELERY_WORKER_CONCURRENCY.

I don't think that sufficiently communicates the situation to a new person.  I ended up having to spend multiple days doing intense research and experimentation to confidently arrive at the conclusions I recorded in my [Celery Setting Primer](https://gist.github.com/odigity/35e36d350884bdaa57998a563b3a735a) gist.

Now I'm in the middle of doing a similar investigation to figure out how `autodiscover_tasks` actually works, because once again, it ain't as simple as the docs pretend.

**Comment by newearthmartin** [comment]
I created another issue to update the documentation https://github.com/celery/celery/issues/7309
